### ISSUE TITLE
Error: Class MotionComponent creator not found

### ISSUE THREAD

**Issue by subdiox** [dava/dava.engine#288]
## Issue
When I open `Data/3d/Maps/hangar_prem_2018/hv1.sc2` in the Data directory of Blitz 4.6, following error occurs:
<img width="532" alt="2018-02-02 18 06 58" src="https://user-images.githubusercontent.com/20740664/35725298-346742dc-0844-11e8-8b0e-949ee5a40794.png">
I think current ResourceEditor does not support new motion introduced in Blitz 4.6.

In addition, I don't know why DAVA Engine hasn't been developed (or committed) since 6 months ago. What happened to DAVA Engine development team?

## Environments
macOS version: 10.13.3
ResourceEditor version: 4.1.0

**Comment by klesch** [comment]
Hello. Let's try again. We had some problems with updating the public repo. And I hope we successfully resolved all of them.

Be sure that you enabled PUBLIC_BUILD at ResourceEditor/CMakeList.txt

**Comment by subdiox** [comment]
Thank you for updating the source code, but I wasn't able to compile it.
In the `_build` folder in `/Programs/ResourceEditor`, I executed the following command:
```
cmake -G "Xcode" -DPUBLIC_BUILD=true -DUNITY_BUILD=true ..
```
And I got the following error after some standard output:
```
CMake Error at CMakeLists.txt:48 (find_package):
  By not providing "FindDavaFramework.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "DavaFramework", but CMake did not find one.

  Could not find a package configuration file provided by "DavaFramework"
  with any of the following names:

    DavaFrameworkConfig.cmake
    davaframework-config.cmake

  Add the installation prefix of "DavaFramework" to CMAKE_PREFIX_PATH or set
  "DavaFramework_DIR" to a directory containing one of the above files.  If
  "DavaFramework" provides a separate development package or SDK, be sure it
  has been installed.


-- Configuring incomplete, errors occurred!
```

I think `/Sources/CMake/Modules` directory does not contain all required cmake files, and this causes the above error.

**Comment by subdiox** [comment]
Could you push the release version of DAVA Engine to release/x.x.x branch (not development branch), if it doesn't bother you?

**Comment by klesch** [comment]
I'll try to fix this issue next week. I think, release branch has same issues as development. 
I'm likely to add some "how to" instructions to build Resource Editor or any other application

**Comment by klesch** [comment]
We updated sources for development branch. I compiled Resource Editor on Mac from this branch with unity_build and public_build options (-DPUBLIC_BUILD=true -DUNITY_BUILD=true) successfully.

The only thing that you have to do is copy dava.engine/Sources/CMake/ConfigureFiles/DavaConfigTemplate.in as dava.engine/DavaConfig.in
and set correct path to QT_PATH variable. We use Qt5.6. Example from my repo:
QT_PATH = /Users/user_name/Work/sdk/Qt/Qt56/clang_64

Let me know if something still not work.

**Comment by subdiox** [comment]
I needed to add execution permission to `Bin/CMakeMac/CMake.app/Contents/bin/cmake`, with the following command: `chmod +x Bin/CMakeMac/CMake.app/Contents/bin/*`
After that, I succeeded in compiling Resource Editor in a way as you said, so I tried to build every tool using `build_tools.py`.
I changed `build_tools.py` a little bit like following:
`values.cmakePath = os.path.join(values.frameworkPath, "Bin/CMake.app/Contents/bin/cmake")` -> `values.cmakePath = os.path.join(values.frameworkPath, "Bin/CMakeMac/CMake.app/Contents/bin/cmake")`
This change solved the error of "File Not Found".
I will send these changes as pull request. Thank you for updating source code to be buildable.
